src/deepqmc/gnn/electron_gnn.py

```diff
@@ -112,14 +113,19 @@ def __init__(
         for lbl in ['w', 'h', 'g', 'u']:
             subnet_factory_by_lbl.setdefault(lbl, subnet_factory)
         if deep_features:
-            self.u = {
-                typ: subnet_factory_by_lbl['u'](
-                    two_particle_stream_dim,
-                    residual=not first_layer,
-                    name=f'u{typ}',
+            self.u = (
+                subnet_factory_by_lbl['u'](
+                    two_particle_stream_dim, residual=not first_layer, name='u'
```

Comment:
How does this work if the streams have different dimensions?

Reply:
Ok, I see. So shared means that we do something along the lines of the concatenation update rule (not sharing parameters, which was my confusion). To me the [code](https://github.com/deepqmc/deepqmc/blob/5f90fd535232f0077759982c335fc79b8a869ed3/src/deepqmc/gnn/electron_gnn.py#L164) associated with this shared update looks a bit cumbersome. If you agree that this is nothing but the concatenate features + MLP (+ split) and the featurewise MLP update rules from the electron stream we should maybe try to come up with a concise unified building block. What do you think?

Reply:
If not combining the updates we should at least align the nomenclature.

Reply:
Ok, confusion is resolved. This actually corresponds to sharing parameters across two particle streams and the case that the two particle streams have different dimensions is not implemented. This could be refactored to be more concise and/or a comment could be added to explain the logic.

Reply:
Would you mind adding this doing this refactor/adding the comment yourself? I think it might be quicker since you know better which aspects you want to improve.